### PR TITLE
Add centos8 cloudtesting scripts

### PR DESCRIPTION
**Pull request by janpriessnitz**
Jenkins cloudtesting job: https://lcgapp-services.cern.ch/cvmfs-jenkins/job/CvmfsCloudTesting/640/CVMFS_PLATFORM=el8,CVMFS_PLATFORM_CONFIG=x86_64,label=trampoline/

### CODE REVIEW

File: test/cloud_testing/platforms/centos8_x86_64_setup.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+. ${script_location}/common_setup.sh
+
+# # Place the overlay directories on ftype=1 16GB xfs partition
+# sudo dd if=/dev/zero of=/xfs-backend bs=$((1024*1024)) count=$((16*1024))
+# sudo mkfs.xfs -n ftype=1 /xfs-backend
+# sudo mkdir -p /var/spool/cvmfs
+# sudo mount /xfs-backend /var/spool/cvmfs
+
+# Place the overlay directories on 16GB ext4 partition
+sudo dd if=/dev/zero of=/ext4-backend bs=$((1024*1024)) count=$((16*1024))
```
Comment (jblomer) [inline test/cloud_testing/platforms/centos8_x86_64_setup.sh:14]: Is this still necessary, i.e. is the default file system xfs on centos?

File: test/cloud_testing/platforms/centos8_x86_64_setup.sh [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+echo "done"
+
+disable_systemd_rate_limit
+
+# Ensure Apache is up and running after package update
+sudo systemctl restart httpd || die "failure in final Apache restart"
+
+# Install repository gateway
+echo "Installing repository gateway"
+package_map=pkgmap.cc7_x86_64
```
Comment (jblomer) [inline test/cloud_testing/platforms/centos8_x86_64_setup.sh:108]: Let's add a `TODO` here for the time when we build the gateway on EL8

File: test/cloud_testing/platforms/centos8_x86_64_test.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+cd ${SOURCE_DIRECTORY}/test
+echo "running CernVM-FS client test cases..."
+CVMFS_TEST_CLASS_NAME=ClientIntegrationTests                                  \
+./run.sh $CLIENT_TEST_LOGFILE -o ${CLIENT_TEST_LOGFILE}${XUNIT_OUTPUT_SUFFIX} \
+                              -x src/005-asetup                               \
+                                 src/004-davinci                              \
+                                 src/007-testjobs                             \
+                                 src/056-lowspeedlimit                        \
+                                 src/084-premounted                           \
```
Comment (jblomer) [inline test/cloud_testing/platforms/centos8_x86_64_test.sh:31]: Shouldn't this work, EL8 comes with fuse3 by default I thought.

File: test/cloud_testing/platforms/centos8_x86_64_test.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+echo "running CernVM-FS server test cases..."
+CVMFS_TEST_CLASS_NAME=ServerIntegrationTests                                  \
+CVMFS_TEST_UNIONFS=overlayfs                                                  \
+./run.sh $SERVER_TEST_LOGFILE -o ${SERVER_TEST_LOGFILE}${XUNIT_OUTPUT_SUFFIX} \
+                              -x src/518-hardlinkstresstest                   \
+                                 src/585-xattrs                               \
+                                 src/600-securecvmfs                          \
+                                 src/628-pythonwrappedcvmfsserver             \
+                                 src/672-publish_stats_hardlinks              \
+                                 src/673-acl                                  \
```
Comment (jblomer) [inline test/cloud_testing/platforms/centos8_x86_64_test.sh:46]: This should also work (fuse3)

File: test/cloud_testing/platforms/centos8_x86_64_test.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                                 src/0*                                       \
+                              || retval=1
+
+
+echo "running CernVM-FS server test cases..."
+CVMFS_TEST_CLASS_NAME=ServerIntegrationTests                                  \
+CVMFS_TEST_UNIONFS=overlayfs                                                  \
+./run.sh $SERVER_TEST_LOGFILE -o ${SERVER_TEST_LOGFILE}${XUNIT_OUTPUT_SUFFIX} \
+                              -x src/518-hardlinkstresstest                   \
+                                 src/585-xattrs                               \
```
Comment (jblomer) [inline test/cloud_testing/platforms/centos8_x86_64_test.sh:42]: I'd think this should also work on overlayfs

File: test/cloud_testing/platforms/centos8_x86_64_setup.sh (hunk 1 above)
Comment (janpriessnitz) [reply test/cloud_testing/platforms/centos8_x86_64_setup.sh:14]: Yes, XFS is still default on Centos 8.

File: test/cloud_testing/platforms/centos8_x86_64_test.sh (hunk 3 above)
Comment (janpriessnitz) [reply test/cloud_testing/platforms/centos8_x86_64_test.sh:31]: I haven't found any fuse3 files on the cc8 openstack image. The packages are available in the repository though.

Comment (jblomer) [reply test/cloud_testing/platforms/centos8_x86_64_test.sh:31]: Ah, I see.  Then we need also two EL8 platforms for the two fuse versions

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest nodestroy

**Pull request merged**